packages/core/src/storages/request_queue_v2.ts

```diff
@@ -215,14 +240,19 @@ export class RequestQueue extends RequestProvider {
                 continue;
             }
 
-            this.queueHeadIds.add(id, id, false);
+            headIdBuffer.push(id);
+            this.assumedForefrontCount = Math.max(0, this.assumedForefrontCount - 1);
             this._cacheRequest(getRequestId(uniqueKey), {
                 requestId: id,
                 uniqueKey,
                 wasAlreadyPresent: true,
                 wasAlreadyHandled: false,
             });
         }
+
+        for (const id of forefront ? headIdBuffer.reverse() : headIdBuffer) {
```

Comment:
Does this reverse matters at all? They're all still gonna be added with forefront so it feels redundant

Reply:
Unfortunately it does, as we want to prepend `headIdBuffer` (as it is) to `queueHeadIds` - if we push the requests one by one in to the head of `queueHeadIds`, it would end up reversed:

```
buffer: 12345
queue: 67890

buffer: 2345
queue: 167890

buffer: 345
queue: 2167890
```

etc. 

I'm not a huge fan of this ternary mumbo-jumbo either, but `headIdBuffer` is at most 25 items long, so at least this should cause no noticeable perf problem. Readability... that's a different story 🥲